In [2]:
import pysam

In [21]:
file = pysam.AlignmentFile("./data/merged-tumor.bam", "rb")
printFields = False

for read in file:
    if not printFields:
        print(read.to_dict())
        print("\n\n" + str(read.flag))
        printFields = True
    else:
        break

file.close()

{'name': 'C0HVYACXX120402:7:1207:5722:57044', 'flag': '1187', 'ref_name': '21', 'ref_pos': '9483249', 'map_quality': '27', 'cigar': '76M', 'next_ref_name': '=', 'next_ref_pos': '9483382', 'length': '209', 'seq': 'TTTTCAAACAGTATCTATGCCTGCCAAATGTGAACATATAAAAAAAAACCAGAATGTGCCATTCTGATTTAAACTG', 'qual': '==<>@?@@>@D?>@C?>8JA5?>CC>?@???BB;H-:4A?D=DB8BDE?GBJCD@B8??E<A>CDJB@B>AA@@@C', 'tags': ['XA:Z:GL000217.1,-110754,76M,1;', 'BD:Z:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN', 'MD:Z:76', 'RG:Z:1', 'BI:Z:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN', 'NM:i:0', 'MQ:i:27', 'AS:i:76', 'XS:i:71']}


1187


[E::idx_find_and_load] Could not retrieve index file for './data/merged-tumor.bam'


<b>Alignment Segment Fields:</b><br/>
<b>NAME</b> &nbsp;&nbsp;&nbsp;(C0HVYACXX120402:7:1207:5722:57044)<br/>
<b>FLAG</b> &nbsp;&nbsp;&nbsp;(1187)<br/>
<b>REF_NAME</b> &nbsp;&nbsp;&nbsp;(21)<br/>
<b>REF_POS</b> &nbsp;&nbsp;&nbsp;(9483249)<br/>
<b>MAP_QUALITY</b> &nbsp;&nbsp;&nbsp;(27)<br/>
<b>CIGAR</b> &nbsp;&nbsp;&nbsp;(76M)<br/>
<b>NEXT_REF_NAME</b> &nbsp;&nbsp;&nbsp;(=)<br/>
<b>NEXT_REF_POS</b> &nbsp;&nbsp;&nbsp;(9483382)<br/>
<b>LENGTH</b> &nbsp;&nbsp;&nbsp;(209)<br/>
<b>SEQ</b>  &nbsp;&nbsp;&nbsp;(TTTTCAAACAGTATCTATGCCTGCCAAATGTGAACATATAAAAAAAAACCAGAATGTGCCATTCTGATTTAAACTG)<br/>
<b>QUAL</b> &nbsp;&nbsp;&nbsp;(==<>@?@@>@D?>@C?>8JA5?>CC>?@???BB;H-:4A?D=DB8BDE?GBJCD@B8??E\<A\>CDJB@B>AA@@@C)<br/>
<b>TAGS:</b><br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- XA:Z:GL000217.1,-110754,76M,1;<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- BD:Z:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- MD:Z:76<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- RG:Z:1<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- BI:Z:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- NM:i:0<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- MQ:i:27<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- AS:i:76<br/>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- XS:i:71<br/>

In [22]:
file = pysam.AlignmentFile("./data/merged-tumor.bam", "rb")
n = 10

for read in file:
    if n > 0:
        n -= 1
        print(read.flag)
    else:
        break

file.close()

1187
163
99
99
99
163
99
1123
99
1123


[E::idx_find_and_load] Could not retrieve index file for './data/merged-tumor.bam'


In [32]:
file = pysam.AlignmentFile("./data/merged-tumor.bam", "rb")
readCount = 0
unmappedCount = 0
zeroQualityCount = 0
avgQuality = 0
avgQualityWithoutZero = 0;

for read in file:
    readCount += 1
    avgQuality += read.mapping_quality
    
    if read.flag & 4 > 0:
        unmappedCount += 1
        
    if read.mapping_quality == 0:
        zeroQualityCount += 1

avgQualityWithoutZero = avgQuality / (readCount - zeroQualityCount)
avgQuality /= readCount

print("Total count of reads: " + str(readCount))
print("Count of unmapped reads: " + str(unmappedCount))
print("Count of reads with mapping quality of 0: " + str(zeroQualityCount))
print("Average mapping quality count: " + str(avgQuality))
print("Average mapping quality without zeros: " + str(avgQualityWithoutZero))
file.close()

[E::idx_find_and_load] Could not retrieve index file for './data/merged-tumor.bam'


Total count of reads: 2921629
Count of unmapped reads: 17765
Count of reads with mapping quality of 0: 126628
Average mapping quality count: 55.91379158681681
Average mapping quality without zeros: 58.446975510921106
